components/pkg-export-docker/src/build.rs

```diff
@@ -84,52 +84,62 @@ fn default_docker_base_image() -> Result<String> {
     /// A string representation of a Habitat Package Identifer for the Habitat CLI package.
-    pub hab:                &'a str,
+    pub hab:                String,
     /// A string representation of a Habitat Package Identifer for the Habitat Launcher package.
-    pub hab_launcher:       &'a str,
+    pub hab_launcher:       String,
     /// A string representation of a Habitat Package Identifer for the Habitat Supervisor package.
-    pub hab_sup:            &'a str,
+    pub hab_sup:            String,
     /// The Builder URL which is used to install all service and extra Habitat packages.
-    pub url:                &'a str,
+    pub url:                String,
     /// The Habitat release channel which is used to install all service and extra Habitat
     /// packages.
     pub channel:            ChannelIdent,
     /// The Builder URL which is used to install all base Habitat packages.
-    pub base_pkgs_url:      &'a str,
+    pub base_pkgs_url:      String,
     /// The Habitat release channel which is used to install all base Habitat packages.
     pub base_pkgs_channel:  ChannelIdent,
     /// A list of either Habitat Package Identifiers or local paths to Habitat Artifact files which
     /// will be installed.
-    pub idents_or_archives: Vec<&'a str>,
+    pub idents_or_archives: Vec<String>,
     /// The Builder Auth Token to use in the request
-    pub auth:               Option<&'a str>,
+    pub auth:               Option<String>,
     /// Base image used in From of dockerfile
     pub base_image:         String,
     /// Whether or not to create an image with a single layer for each
     /// Habitat package.
     pub multi_layer:        bool,
 }
 
-impl<'a> BuildSpec<'a> {
+impl BuildSpec {
     /// Creates a `BuildSpec` from cli arguments.
-    pub fn new_from_cli_matches(m: &'a clap::ArgMatches<'_>, default_url: &'a str) -> Result<Self> {
-        Ok(BuildSpec { hab:                m.value_of("HAB_PKG").unwrap_or(DEFAULT_HAB_IDENT),
+    pub fn new_from_cli_matches(m: &clap::ArgMatches<'_>, default_url: &str) -> Result<Self> {
+        Ok(BuildSpec { hab:                m.value_of("HAB_PKG")
+                                            .unwrap_or(DEFAULT_HAB_IDENT)
+                                            .to_string(),
                        hab_launcher:       m.value_of("HAB_LAUNCHER_PKG")
-                                            .unwrap_or(DEFAULT_LAUNCHER_IDENT),
-                       hab_sup:            m.value_of("HAB_SUP_PKG").unwrap_or(DEFAULT_SUP_IDENT),
-                       url:                m.value_of("BLDR_URL").unwrap_or(&default_url),
+                                            .unwrap_or(DEFAULT_LAUNCHER_IDENT)
+                                            .to_string(),
+                       hab_sup:            m.value_of("HAB_SUP_PKG")
+                                            .unwrap_or(DEFAULT_SUP_IDENT)
+                                            .to_string(),
```

Comment:
I dont know if the effort is worth it, but if we switched this to using `structopt` this duplication could go away. It would also remove duplication in how we define the CLI.

Reply:
Yeah, I'd eventually like to get all our CLI tooling using structopt.